src/Vault.sol

```diff
@@ -566,9 +559,8 @@ contract Vault is IERC4626, ERC20Permit, ILiquidationSource, IClaimable, Ownable
    */
   function mint(uint256 _shares, address _receiver) external virtual override returns (uint256) {
```

Comment:
This function is starting to look *very* similar to the `_depositAssets` function (with exception of the different error names). Since they do the same things, should we use `_depositAssets` in the mint function instead of duplicating the logic?
![image](https://github.com/GenerationSoftware/pt-v5-vault/assets/40277611/501dbd43-5532-4361-90b8-507234ed4c4d)